README.md

```diff
@@ -154,33 +91,107 @@ Host: localhost:6064
+has no explicit configuration, it will simply execute [browserify]. The
+complete output file with the CommonJS require wrapper is exposed to [BFFS].
 
-```
-curl -v http://localhost:1337/cancel/test/0.0.0/dev
+* _Webpack:_ will read the default [webpack] configuration
+(`webpack.config.js`). There are no imposed limitations on the configuration.
+However, the output directory will have to be `./dist` by our convention.
+All files in the output directory will be published to [BFFS].
+
+### Identification of build system type
+
+Specify a build system in `package.json` with the `build` keyword or use
+any of the following terms in the keywords:
 
-GET /cancel/test/0.0.0/dev HTTP/1.1
-Host: localhost:1337
-User-Agent: curl/7.43.0
-Accept: */*
+1. ES6 transpilation: `es2017`, `es2016`, `es2015`, `es6`
+2. Webpack: `webpack`
+3. Browserify: `browserify`
 
-build test@0.0.0 cancelled
+Alternatively specifying the build system name on the `package.json` with
+the relative path to the configuration file will also classify the build
+system, for example: `webpack: '/path/to/config.js'`.
+
+### Forcefully ignore builds
+
+If a published package should not run any builds at all, provide a
+`build: false` flag in the package.json.
+
+```json
+{
+  "name": "package",
+  "version": "1.0.0",
+  "build": false,
+  ...
+}
 ```
 
-### wrhs.toml
+Note: the module/package can also be published directly to a module registry.
+However, if you want to ensure dependents are build whenever your module is
+published this flag can be useful.
+
+### Configuration
+
+Each environment specifies a different set of default options for the builder.
+For instance which registry to run `npm install` against. Each build instance
+has a maximum runtime of `15` minutes. This value can be changed in the
+configuration.
+
+#### Secure setup
+
+By default `carpenterd` runs as an service over `http` and has no
+authentication in place. Setup the configuration to have [Slay] use `https`
+and use authentication middleware, for example [authboot].
+Store API keys and tokens in an encrypted config with [whisper.json][whisper].
+
+#### Per build specifications
+
+Variables and specifications required for a build are discerned from a
+combination of `package.json`, build system configuration files and
+defaults from Carpenters configuration.
+
+**type:** can be supplied as `build` property on the package.json or is
+extracted from the keywords. Defaults to ES6.
+
+**target:** writes the package and its dependencies to a temporary folder
+named after `build.id` a unique v4 `id`. After building this folder is
+removed from the file system to save disk space.
+
+**env:** Retrieved from the package.json `env` property.
+
+**version:** read from the package.json `dist-tags.latest`. Has no default.
+
+**name:** Defaults to the package.json `name` property, e.g. the modules name.
+
+**locale:** Uses the locales specified on the `package.json`. Each unique
```

Comment:
> specified on the `package.json`.

There's going to be a lot to come back through here to clean up once the [configuration changes](https://github.com/warehouseai/extract-config/) are done. cc: @SivanMehta